### ISSUE TITLE
Connections?

### ISSUE THREAD

**Issue by mythicaldog** [rastapasta/pokemon-go-mitm#230]
How many connections will this support? Say I was using it as a backend to provide data to a site allowing anyone to use my endpoint to collect that data, will I run into any issues?

**Comment by trisk** [comment]
pokemon-go-mitm should in theory scale to many thousands of sessions, provided you have enough memory and network resources. This is not documented yet, but you can create request handlers taking three parameters. The first parameter will be an empty object that is unique to the session.

 Be warned that the proxy has not been harded against deliberate abuse, however. There are easy ways to crash it.

**Comment by trisk** [comment]
I suggest modifying the backend to select between different source addresses to avoid hundreds of clients in different physical locations on one address. Alternatively, putting a load balancer in front to route between multiple instances. You probably also want the LB for SSL termination and ensure requests are well formed, maybe also rate limit users.

**Comment by mythicaldog** [comment]
Yeah was going to use Nginx for load balancing. Could you explain how the proxy might be abused and why anyone would want to? Thanks for the information!

**Comment by trisk** [comment]
It's less of an issue if the HTTP proxy component, which is pretty fragile, is not exposed, just the virtual endpoint. Still, someone could forward a lot of invalid requests to the real endpoint to try to get your addresses banned, or attempt to abuse the session tracking (which currently has no ceiling on currency sessions and thus will use up memory for non-expired sessions).

**Comment by mythicaldog** [comment]
I don't need to make any requests at all I'm just reading responses nothing else so I could just disable to proxy all together?

**Comment by trisk** [comment]
Huh? The proxy (whether as an HTTP proxy or virtual endpoint) receives real requests from the app and forwards them to the server, and forwards he responses back to the app. That's the only way to see the traffic via MITM. If you allow anyone to connect to it, they may send all sorts of garbage.

**Comment by trisk** [comment]
If you mean the HTTP proxy, there's no way to disable it separately atm, but if it's on a private network behind load balancers no one can connect to it anyway.
